compiler-core/src/type_/tests.rs

```diff
@@ -118,8 +118,9 @@ fn get_warnings(src: &str, deps: Vec<DependencyModule<'_>>) -> Vec<Warning> {
     warnings
         .take()
         .into_iter()
-        .map(|warning| match warning {
-            crate::Warning::Type { warning, .. } => warning,
+        .filter_map(|warning| match warning {
+            crate::Warning::Type { warning, .. } => Some(warning),
+            _ => None,
```

Comment:
Could you match on all the variants rather than using a catch all please? That way the compiler will prompt us to update this code when new ones are added. Outdated logic from catch-all patterns has been a common source of bugs in the past.